apps/frontend/src/pages/CreateListing.tsx

```diff
@@ -1,5 +1,281 @@
+        longitude: 0.0,
+      },
+      images: [],
+    },
+  });
+
+  // Gets the user location, and adds it to the listing object
+  const getUserLocation = useCallback(async () => {
+    try {
+      navigator.geolocation.getCurrentPosition((position) => {
+        newListingObject.listing.location.longitude = position.coords.longitude;
+        newListingObject.listing.location.latitude = position.coords.latitude;
+      });
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }, [newListingObject.listing.location]);
+
+  // Updates and sends the newListingObject, to the server via post under /api/listing
+  const sendPostToCreateListing = async () => {
+    // In order to make sure that the images are retrieved before submitting
+    const successImages = await uploadImages();
+    let successLocation: boolean = true;
+    if (newListingObject.listing.location.latitude !== 0) {
+      successLocation = false;
+      getUserLocation().then(() => {
+        successLocation = true;
+      });
+    }
+    //console.log("Image Successful", imageUploadSuccessful)
+    console.log("New Listing Object", newListingObject);
+    if (successImages && successLocation) {
+      _axios_instance
+        .post("/listing", newListingObject)
+        .then((response) => {
+          alert("Listing Created!");
+          console.log("Response to upload", response);
+        })
+        .catch((error) => {
+          console.log("Response to error", error);
+          alert("Listing Creation Failed");
+        });
+    } else if (!successImages) {
+      alert("Images failed to upload, please try again later");
+    } else {
+      // Currently this is added to catch if the location is not set, we could default this to the location of the university instead
+      alert(
+        "Error occurred when creating a listing, you may need to enable location permissions for this site",
+      );
+    }
+  };
+
+  // This function makes sure that the passed in url is a base64 data string
+  const isImageValid = (url: string) => {
+    try {
+      return url.startsWith("data");
+    } catch (error) {
+      return error === TypeError && url === undefined;
+    }
+  };
+
+  // Request the user's location on load
+  useEffect(() => {
+    getUserLocation().then((r) => console.log(r));
+  }, [getUserLocation]);
+
+  // Uploads the images to the s3 server, this is handled seperately
+  // This may need to be modified to use File objects instead of the file strings
+  const uploadImages = async () => {
+    try {
+      await Promise.all(
+        images.map(async (image) => {
+          try {
+            const response = await _axios_instance.post(
+              "/images",
+              { image: image },
+              {
+                headers: {
+                  "Content-Type": "multipart/form-data",
+                },
+              },
+            );
+            newListingObject.listing.images.push({ url: response.data.url });
```

Comment:
newListingObject should be updated with useState

Reply:
Yeah, I tried that for a long time, but it never seems to work, I had tried to reassign images to a list of `imageUrls = [{url: "www.example.com/image.png"},...] ` but it never actually updated the newListingObject. Is there another way to be updating that value, or maybe my code is just wrong?

```ts
setNewListingObject((prevState) => ({
          ...prevState,
          listing: {
            ...prevState.listing,
            images:  imageUrls
            },
          },
        }));
```

Reply:
That should work as far as I can tell. We can take a look at it together in the lab today